### PR TITLE
lots of small changes

### PR DESCRIPTION
**Pull request by fanick-ber**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -5,4 +5,6 @@ DATABASE_USERNAME=username
 DATABASE_PASSWORD=secretpassword
 DATABASE_NAME=dbname
 JWT_SECRET=jwt_awesome_key
+JWT_SECRET_KEY=jwt_awesome_key
+JWT_EXPIRATION_TIME=3600
```
Comment (fanick-ber) [inline .env]: that was somehow needed to get the jwt running.

The `JWT_SECRET` and `JWT_SECRET_KEY` needed to co-exist for whatever reason.

File: .env.development [hunk 2]
```diff
@@ -5,4 +5,6 @@ DATABASE_USERNAME=username
 DATABASE_PASSWORD=secretpassword
 DATABASE_NAME=dbname
 JWT_SECRET=jwt_awesome_key
-API_PORT=3000
+JWT_SECRET_KEY=jwt_awesome_key
+JWT_EXPIRATION_TIME=3600
+API_PORT=3001
```
Comment (fanick-ber) [inline .env.development]: probably should not be here. I had the docker-compose running in parallel to the direct execution. Different ports could be helpful.

File: src/modules/auth/auth.controller.ts [hunk 3]
```diff
@@ -1,26 +1,25 @@
-import {Body, Controller, HttpStatus, Post, UseGuards} from '@nestjs/common';
-import {ApiResponse, ApiTags} from '@nestjs/swagger';
-import {AuthGuard} from '@nestjs/passport';
+import { Body, Controller, HttpStatus, Post, UseGuards } from '@nestjs/common';
```
Comment (fanick-ber) [inline src/modules/auth/auth.controller.ts]: sorry for the formatting changes. I deactivated my prettier.

File: src/modules/auth/auth.controller.ts [hunk 4]
```diff
@@ -1,26 +1,25 @@
...
-  ) {
-  }
+  ) {}
 
   @Post('login')
-  @UseGuards(AuthGuard())
-  @ApiResponse({status: HttpStatus.ACCEPTED, description: 'Successful Login'})
-  @ApiResponse({status: HttpStatus.BAD_REQUEST, description: 'Bad Request'})
-  @ApiResponse({status: HttpStatus.UNAUTHORIZED, description: 'Unauthorized'})
+  // @UseGuards(AuthGuard())
```
Comment (fanick-ber) [inline src/modules/auth/auth.controller.ts]: The login route should not be protected?

File: src/modules/auth/auth.service.ts [hunk 5]
```diff
@@ -1,28 +1,30 @@
...
     private readonly jwtService: JwtService,
     private readonly configService: ConfigService,
     private readonly userService: UsersService,
-  ) {
-  }
+  ) {}
 
   async generateToken(user: User) {
+    const userPayload = user;
+    delete userPayload.password;
```
Comment (fanick-ber) [inline src/modules/auth/auth.service.ts]: password should not be in plain text in the token (only plain on register then, not on login)

File: src/modules/user/user.service.ts [hunk 6]
```diff
@@ -29,8 +28,8 @@ export class UsersService {
   }
 
   async getByEmailAndPass(email: string, password: string) {
-    const passHash = createHmac('sha256', password).digest('hex');
-    return await this.userRepository.findOne({email, password: passHash});
+    // const passHash = createHmac('sha256', password).digest('hex');
```
Comment (fanick-ber) [inline src/modules/user/user.service.ts]: the password transform takes already care of that. Did work without the additional hashing.

File: .env.development (hunk 2 above)
Comment (papotte) [reply .env.development]: Yes, here it works :D

File: src/modules/auth/auth.controller.ts (hunk 3 above)
Comment (papotte) [reply src/modules/auth/auth.controller.ts]: Should we create some Linting rules and go with that?

**Pull request merged**